virttest/utils_disk.py

```diff
@@ -76,32 +77,42 @@ def is_mount(src, dst=None, fstype=None, options=None, verbose=False, session=No
 
     :return: True if mounted, else return False
     """
-    mount_str = "%s %s %s" % (src, dst, fstype)
-    mount_str = mount_str.replace("None", "").strip()
-    mount_list_cmd = "cat /proc/mounts"
-
-    if session:
-        mount_result = session.cmd_output_safe(mount_list_cmd)
-    else:
-        mount_result = process.run(mount_list_cmd, shell=True).stdout_text
-    if verbose:
-        LOG.debug("/proc/mounts contents:\n%s", mount_result)
-
-    for result in mount_result.splitlines():
-        if mount_str in result:
-            if options:
-                options = options.split(",")
-                options_result = result.split()[3].split(",")
-                for op in options:
-                    if op not in options_result:
-                        if verbose:
-                            LOG.info(
-                                "%s is not mounted with given" " option %s", src, op
-                            )
-                        return False
-            if verbose:
-                LOG.info("%s is mounted", src)
-            return True
+    command_parts = [
+        part
+        for part in [
+            f"-S {src}" if src is not None else None,
+            f"-T {dst}" if dst is not None else None,
+            f"-t {fstype}" if fstype is not None else None,
+        ]
+        if part is not None
+    ]
```

Comment:
The code is quite complex, wrap twice. How about simpler to 
```python
options = [
    ("-T", dst),
    ("-t", fstype),
]
mount_opts = " ".join(f"{opt} {val}" for opt, val in options if val is not None)
mount_check_cmd = f"findmnt -J -S {src} {mount_opts}"
```

Reply:
OKay, both are fine for me.Thanks.

Reply:
Updated the code, thanks.